test/services/ldap/hooks/restrictToSchoolSystems.test.js

```diff
@@ -0,0 +1,76 @@
+const { expect } = require('chai');
+const { BadRequest, Forbidden } = require('@feathersjs/errors');
+const { ObjectId } = require('mongoose').Types;
+const fut = require('../../../../src/services/ldap/hooks/restrictToSchoolSystems');
+
+describe('restrictToSchoolSystems', () => {
+	it('rejects requests that don\'t meet the requirements', () => {
+		try {
+			fut({});
+			throw new Error('This should never happen');
+		} catch (err) {
+			expect(err).to.be.instanceOf(BadRequest);
+			expect(err.message).to.equal('Unexpected call to restrictToValidSystems.');
+		}
+		try {
+			fut({ id: (new ObjectId()).toString() });
+			throw new Error('This should never happen');
+		} catch (err) {
+			expect(err).to.be.instanceOf(BadRequest);
+			expect(err.message).to.equal('Unexpected call to restrictToValidSystems.');
+		}
+		try {
+			fut({ params: { school: { systems: [] } } });
+			throw new Error('This should never happen');
+		} catch (err) {
+			expect(err).to.be.instanceOf(BadRequest);
+			expect(err.message).to.equal('Unexpected call to restrictToValidSystems.');
+		}
+	});
+
+	it('should forbid access to systems not in use by the current school', () => {
+		try {
+			fut({
+				id: (new ObjectId()).toString(),
+				params: {
+					school: {
+						systems: [new ObjectId(), new ObjectId()],
+					},
+				},
+			});
+			throw new Error('This should never happen');
+		} catch (err) {
+			expect(err).to.be.instanceOf(Forbidden);
+		}
+	});
+
+	it('should forbid access if the curretn school has no system', () => {
```

Comment:
small typo in test title

Reply:
You were the last person I expected to find that 😁

Reply:
fixed